server/main.cpp

```diff
@@ -104,6 +112,165 @@ int create_listen_socket()
 
 	return fd;
 }
+
+int get_service_pid(const std::string& service_name, pid_t& pid) {
```

Comment:
now that "application" code is growing, it may be spun off in a new file "start_application" or something.